demos/general/demo_OnACID.py

```diff
@@ -1,104 +1,115 @@
-# -*- coding: utf-8 -*-
 
 """
 Basic demo for the CaImAn Online algorithm (OnACID) using CNMF initialization.
 It demonstrates the construction of the params and online_cnmf objects and
 the fit function that is used to run the algorithm.
 For a more complete demo check the script demo_OnACID_mesoscope.py
-
-@author: jfriedrich & epnev
 """
-#%%
-from IPython import get_ipython
+
+import argparse
+#import code
 import logging
 import numpy as np
 import os
 
+try:
+    cv2.setNumThreads(0)
+except:
+    pass
+
 import caiman as cm
 from caiman.source_extraction import cnmf as cnmf
 from caiman.paths import caiman_datadir
 
-try:
-    if __IPYTHON__:
-        print("Detected iPython")
-        ipython = get_ipython()
-        ipython.run_line_magic('load_ext', 'autoreload')
-        ipython.run_line_magic('autoreload', '2')
-        ipython.run_line_magic('matplotlib', 'qt')
-except NameError:
-    pass
 
-#%%
-# Set up the logger; change this if you like.
-# You can log to a file using the filename parameter, or make the output more or less
-# verbose by setting level to logging.DEBUG, logging.INFO, logging.WARNING, or logging.ERROR
-
-logging.basicConfig(format=
-                    "%(relativeCreated)12d [%(filename)s:%(funcName)20s():%(lineno)s]"\
-                    "[%(process)d] %(message)s",
-                    level=logging.WARNING)
-    # filename="/tmp/caiman.log"
-# %%
 def main():
-    pass  # For compatibility between running under an IDE and the CLI
+    cfg = handle_args()
+
+    if cfg.logfile:
+        logging.basicConfig(format=
+            "%(relativeCreated)12d [%(filename)s:%(funcName)20s():%(lineno)s][%(process)d] %(message)s",
+            level=logging.WARNING,
+            filename=cfg.logfile)
+        # You can make the output more or less verbose by setting level to logging.DEBUG, logging.INFO, logging.WARNING, or logging.ERROR
+    else:
+        logging.basicConfig(format=
+            "%(relativeCreated)12d [%(filename)s:%(funcName)20s():%(lineno)s][%(process)d] %(message)s",
+            level=logging.WARNING)
 
-    # %% load data
-    fname = [os.path.join(caiman_datadir(), 'example_movies', 'demoMovie.tif')]
+    if cfg.input is None:
+        fnames = [os.path.join(caiman_datadir(), 'example_movies', 'demoMovie.tif')]
+    else:
+        fnames = cfg.input
 
-    # %% set up some parameters
-    fr = 10  # frame rate (Hz)
-    decay_time = .75  # approximate length of transient event in seconds
-    gSig = [6, 6]  # expected half size of neurons
-    p = 1  # order of AR indicator dynamics
-    min_SNR = 1  # minimum SNR for accepting candidate components
-    thresh_CNN_noisy = 0.65  # CNN threshold for candidate components
-    gnb = 2  # number of background components
-    init_method = 'cnmf'  # initialization method
+    if cfg.configfile:
+        opts = cnmf.params.CNMFParams(params_from_file=cfg.configfile)
+        if opts.data['fnames'] is None:
+            opts.set("data", {"fnames": fnames})
```

Comment:
And finally we reach (the start of) one of the highlights of all this work: JSON-based configs for CLI tools.